### PR TITLE
mes-8785: Amend query to filter when rekey is provided

### PR DESCRIPTION
**Pull request by andrewsetterfield**
# Description and relevant Jira numbers

## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: src/functions/searchResults/framework/database/query-builder.ts [hunk 1]
```diff
@@ -5,6 +5,10 @@ export const getConciseSearchResultsFromSearchQuery = (queryParameters: QueryPar
   const parameterArray: string[] = [];
   let queries: string[] = [];
   let queryString: string = '';
+  let rekeyFlag: boolean = false;
+
+  // Require a staff number to allow rekeyflag to be used
+  if (queryParameters?.staffNumber && queryParameters?.rekey === 'true') rekeyFlag = true;
```
Comment (matthew2564) [inline src/functions/searchResults/framework/database/query-builder.ts]: Do we need a staffNumber check here seeing as we are checking via the QueryParams and defaulting to false if omitted?

Comment (andrewsetterfield) [reply src/functions/searchResults/framework/database/query-builder.ts]: will make a few changes, no need to do things twice

File: src/functions/searchResults/framework/handler.ts [hunk 2]
```diff
@@ -42,6 +41,12 @@ export async function handler(event: APIGatewayEvent): Promise<Response> {
     if (event.queryStringParameters.staffNumber) {
       queryParameters.staffNumber = event.queryStringParameters.staffNumber;
     }
+    // guard against filtering on rekey without providing a staff number
+
+    if (event.queryStringParameters.rekey) {
+      queryParameters.rekey = !!(event.queryStringParameters.rekey === 'true' && queryParameters.staffNumber);
+      console.log('queryParameters.rekey', queryParameters.rekey);
```
Comment (matthew2564) [inline src/functions/searchResults/framework/handler.ts]: Remove this

File: src/functions/searchResults/framework/database/query-builder.ts [hunk 3]
```diff
@@ -63,13 +63,21 @@ export const getConciseSearchResultsFromSearchQuery = (queryParameters: QueryPar
...
+    queryString = queryString.concat('SELECT test_result FROM TEST_RESULT WHERE ');
+  }
 
   queries.forEach((query) => {
     queryString = queryString.concat(`${query} `);
   });
 
-  queryString = queryString.concat('ORDER BY test_date DESC LIMIT 200;');
+  // If rekeyFlag apply the filter for rekey first prior to restricting number of records
+  if (queryParameters.rekey) {queryString = queryString.concat(
```
Comment (matthew2564) [inline src/functions/searchResults/framework/database/query-builder.ts]: Can you drop down the `queryString` onto it's own line for readability please

**Review by matthew2564 — APPROVED** [review]

**Review by RLCorp — APPROVED** [review]

**Pull request merged**